UnityProject/Assets/Scripts/Health/Objects/Integrity.cs

```diff
@@ -125,6 +126,7 @@ public class Integrity : NetworkBehaviour, IHealth, IFireExposable, IRightClicka
 	private RegisterTile registerTile;
 	public RegisterTile RegisterTile => registerTile;
 	private IPushable pushable;
+	public Meleeable Meleeable => GetComponent<Meleeable>();
```

Comment:
it is better to cache this reference. Like it is right now it will call GetComponent every time the ``Meleeable`` property is used anywhere in the consumer code. Yeah, currently it is used in ``WhenDestroyed`` from ``ConstructibleDoor`` component, so we won't notice it until it becomes a problem later in the future.